titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/validator/ParallelValidator.java

```diff
@@ -0,0 +1,78 @@
+package com.netflix.titus.api.jobmanager.model.job.validator;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.common.model.validator.EntityValidator;
+import com.netflix.titus.common.model.validator.ValidationError;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuples;
+
+/**
+ * A ParallelValidator executes multiple {@link EntityValidator}s in parallel.
+ */
+public class ParallelValidator implements EntityValidator<JobDescriptor> {
+    private final Duration timeout;
+    private final Collection<EntityValidator<JobDescriptor>> hardValidators;
+    private final Collection<EntityValidator<JobDescriptor>> softValidators;
+
+    /**
+     * A ParallelValidator ensures that all its constituent validators complete within a specified duration.  The
+     * distinction between "hard" and "soft" validators is behavior on timeout.  When a "hard" validator fails to
+     * complete within the specified duration it produces a {@link ValidationError} with a {@link ValidationError.Type}
+     * of HARD.  Conversely "soft" validators which fail to complete within the specified duration have a
+     * {@link ValidationError.Type} of SOFT.
+     *
+     * @param timeout The duration within which each validator should complete.
+     * @param hardValidators Those validators which should generate a HARD {@link ValidationError} on timeout.
+     * @param softValidators Those validators which should generate a SOFT {@link ValidationError} on timeout.
+     */
+    public ParallelValidator(
+            Duration timeout,
```

Comment:
Presumably this would eventually move to a ValidatorConfiguration class in the future?

Reply:
Yes.  I'll be figuring out the injection of a ParallelValidator with appropriate internal validators in a subsequent PR.